### ISSUE TITLE
Fix when loading with Mono or Wine

### ISSUE THREAD

**Issue by Tcll** [libertyernie/brawltools#217]
looks like the issue with Brawlbox not seeing `./Lib` is still a thing
The thing I did to get it working was literally something a caveman would do
![image](https://user-images.githubusercontent.com/773959/54385513-2bb25d80-466d-11e9-8839-8afaa5ab4481.png)
just copy the files from `./Lib/` into `./` (3rd run above):
![image](https://user-images.githubusercontent.com/773959/54385538-3ec52d80-466d-11e9-8755-2d5d07a87a85.png)
As for it actually loading files, that has yet to be tested

note that this IS the hotfix-1 version (latest at this time)
installed on Xubuntu 16.04
- Wine 4.0
- Mono  4.2.1

**Comment by Tcll** [comment]
My personal fix is to just use the previous version:
![image](https://user-images.githubusercontent.com/773959/54386938-87cab100-4670-11e9-9bf5-0be95144277d.png)

**Comment by Sammi-Husky** [comment]
@Tcll When loading, are you keeping the app.config file in the same directory as the brawlbox executable? That config _should_ be telling it to look in ./lib and as far as i could find should be a mono compatible solution.

**Comment by Tcll** [comment]
yes, I haven't touched anything but the Lib directory when it comes to moving files

however, I did rename `./BrawlBox.exe` to `./BB078.exe` since I still use WinXP in a net-less VM for various reasons (file association between 2 apps with the same name is a B).
perhaps renaming the config file would help here :P
I'll have to test this whenever I get around to reinstalling Wine as a boot loader update broke my partition order and I had to reinstall.

honestly though, can't you just hard-code this without a config file :/
or at the very least store the config file in `./config/` and **not** make it reliant on the app name (if it is).

I understand the ability to have `./Lib/` in another location, and very much support the option ;)
but tieing the config name to the app name is asking for issues.

**Comment by Sammi-Husky** [comment]
It isn't a conscious decision to tie the config to the app name, it's just the way that .net works. When using a config, .net **requires** it be named the same as the executable.

The recommended and default way for specifying library load locations for .Net stuff is the app.config, so the mechanism isn't something we did ourselves. We probably can manually load the libraries from ./lib ourselves via some C# code but it's frowned upon to do so for a few reasons.

So we're are in situation now where we decide which option is the lesser of the two evils: (reverting to having the DLL's in the same folder, or requiring the config name match)

**Comment by Tcll** [comment]
why would it be frowned upon :/
just write a wrapper to load them and separate the src so it's not so dirty...

Of course, this is coming from the guy who'd rather throw away the python standard library and write his own better modules using the smallest amount of necessary binary modules he can get away with while butchering any PEP8 standards and being the most hated person in the python community.

yeah... I'd do it. :awesomeface:
If it makes the program work better, I'm all for it.

Unless it's about as shoddy as SciCraft farms which break nearly every Minecraft update...
(use raw functionality and you shouldn't have to worry about that)
